Feature/android nativesdk integration

elefantel opened the pull request:
<!--
Thanks for your contribution!

Please ensure that any applicable requirements below are satisfied before submitting this pull request. This will help ensure a quick and efficient review cycle.
-->

**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/coding_guidelines/CODING_GUIDELINES.md)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add the appropiate QA label when dev review is completed
    - `needs-qa`: PR requires manual QA.
    - `No QA/E2E only`: PR does not require any manual QA effort. Prior to merging, ensure that you have successful end-to-end test runs in Bitrise. 
    - `Spot check on release build`: PR does not require feature QA but needs non-automated verification. In the description section, provide test scenarios. Add screenshots, and or recordings of what was tested.
5. Add `QA Passed` label when QA has signed off (Only required if the PR was labeled with `needs-qa`)
6. Add your team's label, i.e. label starting with `team-` (or `external-contributor` label if your not a MetaMask employee)

**Description**
This PR integrates the Android MetaMask SDK. Unlike the rest of the SDKs which use socket.io and interact with the existing communication layer, the Android SDK uses Inter Process Communication (IPC) [implemented via the Android Interface Definition Language (AIDL)] between the dapp and MetaMask mobile.

This requires the addition of a new communication layer as the existing one is catered for socket.io. Because this new layer depends on react-native, it necessitated being added on mobile rather than in the `sdk-communication-layer`. 

The Android communication layer is implemented in `AndroidService` and uses `AndroidSDKEventHandler` for handling events from the Android Native Module (added as an `nativesdk.aar` dependency). The native module then communicates with the dapp through IPC.

**Screenshots/Recordings**
https://github.com/MetaMask/metamask-mobile/assets/17785504/789083fa-3926-4c02-b29c-8072a71e2ed0

**Issue**

fixes #???

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [x] Any added code is fully documented

I have read the CLA Document and I hereby sign the CLA

elefantel committed:
Add native Android SDK dependency

elefantel committed:
Add native Android SDK dependency

elefantel committed:
Add SDK event handlers

elefantel committed:
Add SDK event handlers

elefantel committed:
Add ecies lib dependency

elefantel committed:
Add ecies lib dependency

elefantel committed:
Add SDK connect and accounts request logic in AndroidService

elefantel committed:
Add SDK connect and accounts request logic in AndroidService

elefantel committed:
Add request handling on AndroidService

elefantel committed:
Add request handling on AndroidService

elefantel committed:
Add AndroidServicePort and update native module lib

elefantel committed:
Add AndroidServicePort and update native module lib

elefantel committed:
Remove logging

elefantel committed:
Remove logging

elefantel committed:
AndroidService integration with SDKConnect

elefantel committed:
AndroidService integration with SDKConnect

abretonc7s committed:
feat: temp commit for metamask

abretonc7s committed:
feat: temp commit for metamask

abretonc7s committed:
feat: working state

abretonc7s committed:
feat: working state

Gudahtt committed:
refactor: Simplify `handleNetworkSwitch` helper (#6878)

The `handleNetworkSwitch` helper function no longer requires the
`frequentRpcList` state as a parameter. Instead it takes a reference to
the `PreferencesController`, and gets the RPC list state from that.

Typically it would be preferrable to reference controller state from
Redux rather than accesing it directly from the Engine global. However,
this function will be called from non-UI contexts, so it has to get
controller state from global context regardless.

This change was done to simplify PR #6872, which is part of
https://github.com/MetaMask/mobile-planning/issues/798

Gudahtt committed:
refactor: Consistent provider config naming (#6880)

The `providerConfig` property of the `NetworkController` used to be
named `provider`, which was very easily confused with the `provider`
instance (and with various other things that we call "provider"). When
that renaming happened, not all local variables and function names were
updated accordingly.

All references to `providerConfig` now use the term "providerConfig".

This change was done to simplify PR #6872, which is part of
https://github.com/MetaMask/mobile-planning/issues/798

sethkfman committed:
ci(builds): Sentry sourcemap deploy (#6813)

* updated bitrise script and build sh to only ship sourcemaps on release to store builds

* update build_android_release step

* update build_android_release step

SamuelSalas committed:
feat: Create a performance E2E test for warm starts (#6865)

* Feature file create to test the warm start of the app. Updated Selectors from getElementByPlatform to getXpathElementByResourceId method

* Files Updated to run the Warm start feature on multiple devices

* Move single step from warm start performance to app launch times. Added foreground step to the Warm Start Performance feature

* Update Wallet screen wait method. Update the expected launch time to 8 seconds

* Update the cucumber tag to performance

abretonc7s committed:
feat: working android service

abretonc7s committed:
feat: working android service

github-actions[bot] committed:
7.3.1 (#6912)

* 7.3.0

* update provisioning profile for builds

* Hide approval flow loader if pending approval (#6767)

* fix: 7.3.0 - update terms of use metrics (#6761)

* fix: update cookie-tough dependency (#6772)

* adjust steps for UI changes (#6768)

* fix: update snapshot for linea mainnet (#6790)

* build 1144 (#6793)

* 7.3.1

* fix: invalid transaction data used for approve transaction (#6833) (#6914)

* use valid txn data when approving txn

* rename variable

Co-authored-by: Sylva Elendu <iamsylvaelendu@gmail.com>

* update changelog for release

---------

Co-authored-by: metamaskbot <metamaskbot@users.noreply.github.com>
Co-authored-by: sethkfman <Seth.Kaufman@consensys.net>
Co-authored-by: Matthew Walsh <matthew.walsh@consensys.net>
Co-authored-by: tommasini <46944231+tommasini@users.noreply.github.com>
Co-authored-by: Chris Wilcox <chris.wilcox@consensys.net>
Co-authored-by: sethkfman <10342624+sethkfman@users.noreply.github.com>
Co-authored-by: Sylva Elendu <iamsylvaelendu@gmail.com>

cortisiko committed:
docs: update PR template to include QA labels. (#6862)

* update PR template to include QA labels.

* update NO QA label description

Cal-L committed:
fix: #893 fix immediate lock timer (#6653)

* Stage saga changes

* Create action creators for authentication state machine

* Remove unused code

* Start and end state machine

* Clean up LockScreen

* Clean up LockManager

* Remove unused code

* Notify auth state machine from Authentication service

* Organize auth state machine in sagas

* Change sagas to TS

* Remove unused changes

* Add type to navigation in NavigationService

* Clean up sagas

* Remove console log

* Add tests for sagas

* Remove password set

* Fix saga lint issues

* Fix missing deps

* Remove ts ignore from sagas

* Clarify navigation action

* Remove commented logger

* Create LOCK_SCREEN route const

* Provide comments to sagas

* Rename biometrics listener

* set timer immediately v2

* Add biometricsStateMachineId to auth code

* Remove unused user actions

* Clean up LockManager logic to work with Android and iOS

* Better handle authentication sagas

* Fix saga export error

* Fix saga unit tests

* Bump version for test

* Refine logout state when using biometrics

* Bump version

* Remove retry logic for lock screen

* Cancel biometrics state machine when logging out

* Clean up lock screen

* Reset isAuthenticating when biometrics fail in SecureKeychain.js

* Move LockScreen to top nav level

* Persist app state when LockManager shows.

* Fix saga tests

* Move LockManager initialization

* Use existing login and logout actions for biometric state machine

* Bump versions

* Use replace when logging out

* Update LockScreen snapshoit

* Comment bioStateMachineId

* Remove LockManager to LockManagerService

* Add comments to Authentication service

* Convert LockManagerService to Typescript

* Fix typo

* Update LockManagerService visibility modifiers

* Fix lint issues

---------

Co-authored-by: tommasini <46944231+tommasini@users.noreply.github.com>
Co-authored-by: tommasini <tommasini15@gmail.com>

github-actions[bot] committed:
7.4.0 (#6806)

* 7.4.0

* updated format changelog

* fix: define ReactNativePayments podspec path in podfile (#6821)

* from address changing on transaction object (#6841)

* fix: invalid transaction data used for approve transaction (#6833) (#6842)

* use valid txn data when approving txn

* rename variable

Co-authored-by: Sylva Elendu <iamsylvaelendu@gmail.com>

* add check approval request exists before rejecting (#6838)

* build 1155 RC2 (#6861)

* fix(test): Permission Systems fix taping connect modals (#6868)

* fix tapping connect modals

* fixed lint issues

* removed unused function, not referenced anywhere

* chore: merge 7.3.1 cherry-pick (#6923)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

* chore: version bump for 7.4.0 RC 3 (#6924)

* build 1159 RC4 (#6928)

---------

Co-authored-by: metamaskbot <metamaskbot@users.noreply.github.com>
Co-authored-by: sethkfman <Seth.Kaufman@consensys.net>
Co-authored-by: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Co-authored-by: tommasini <46944231+tommasini@users.noreply.github.com>
Co-authored-by: sethkfman <10342624+sethkfman@users.noreply.github.com>
Co-authored-by: Sylva Elendu <iamsylvaelendu@gmail.com>
Co-authored-by: Vinicius Stevam <45455812+vinistevam@users.noreply.github.com>
Co-authored-by: Chris Wilcox <chris.wilcox@consensys.net>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump word-wrap from 1.2.3 to 1.2.4 (#6840)

Bumps [word-wrap](https://github.com/jonschlinkert/word-wrap) from 1.2.3 to 1.2.4.
- [Release notes](https://github.com/jonschlinkert/word-wrap/releases)
- [Commits](https://github.com/jonschlinkert/word-wrap/compare/1.2.3...1.2.4)

---
updated-dependencies:
- dependency-name: word-wrap
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: sethkfman <10342624+sethkfman@users.noreply.github.com>
Co-authored-by: legobeat <109787230+legobeat@users.noreply.github.com>

Gudahtt committed:
chore: remove unused Jest preprocessor (#6913)

The file `jest.preprocessor.js` has been unused since #6170.

gantunesr committed:
docs: Update QA section in PR template (#6930)

SamuelSalas committed:
fix: Remove splash animation wait step from Cold Start Launch time test script (#6931)

* Change wait for splash animation to expect to displayed assertion

* Remove splash animation steps

abretonc7s committed:
feat: working sdk state

abretonc7s committed:
feat: working sdk state

tommasini committed:
feat: Translation (#6937)

abretonc7s committed:
feat: initialize binding from android service

abretonc7s committed:
feat: initialize binding from android service

elefantel committed:
Always bind Android service when receiving bind deeplink

elefantel committed:
Always bind Android service when receiving bind deeplink

elefantel committed:
Update native module

elefantel committed:
Update native module

vinistevam committed:
remove onBackdropPress for signatures (#6936)

sleepytanya committed:
feat: Migrate wdio 'SendEthMultisig' test to Detox (#6934)

gauthierpetetin committed:
feat: github action to check if PR has requested labels before being merged (#6794)

* feat(action): check if pr includes requested labels

* fix(template): adjust PR temaple

* fix(action): add missing QA label

* fix(action): check if no label prevents merging

* fix(action): remove QA label check

* fix(action): add missing reopened condition

* fix(action): increase list of labels which prevent merges

* fix(action): add labeling guidelines

* fix(action): typo

* fix(action): typo

elefantel committed:
Clean up logs

elefantel committed:
Clean up logs

sleepytanya committed:
fix: update README.md (#6853)

sleepytanya committed:
Restore deleted SendEthMultisig.feature file (#6956)

brianacnguyen committed:
fix: custom position logic for badgeWrapper (#6864)

* Fixed custom position logic for badgeWrapper

* Fixed lint error

brianacnguyen committed:
Added README to Avatar (#6895)

* Added README to Avatar

* Updated readme based on comments

* Fixed reference link for AvatarAccount

brianacnguyen committed:
style: Updated minor checkbox style (#6942)

* Updated minor checkbox changes

* Updated MultiSelect snapshot

vinistevam committed:
feat: add support for template and header on result pages (#6832)

sleepytanya committed:
Fix: add missed parenheses in the function call (#6958)

tommasini committed:
translation ens ipfs alert (#6966)

mmenoxx committed:
chore: small refactor (#6963)

* small refacto

* fix(): fixed linter errors

* fix(): fixed linter errors

---------

Co-authored-by: sethkfman <10342624+sethkfman@users.noreply.github.com>

segun committed:
feat: blockaid what's new popup (#6901)

* Add blockaid what's new

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* New image for both dark/light mode

* image should be first
fix podfile

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* What's new fix navigation

* Revert Podfile

* use Routes

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* revert podfile.lock

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* Show what's new only if blockaid is enabled in settings.

---------

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

segun committed:
feat: Add security alerts settings to experimental tab (#6779)

* Add security alerts settings to experimental tab

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* add more providers label

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* change styles label

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* Fix tests

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* Sort imports

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* use environment var to determine if to show blockaid or not

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* Removed Podfile changes

---------

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

sleepytanya committed:
fix: revert PR 6958 (#6975)

Cal-L committed:
fix: Fix TS errors in `component-library` directory (#6947)

* Fix snapshots

* Clean up hitSlop resolution

* Fix CellBase types reference error

---------

Co-authored-by: LeoTM <1881059+leotm@users.noreply.github.com>

elefantel committed:
Merge branch 'feature/android-nativesdk-integration' of github.com:MetaMask/metamask-mobile into feature/android-nativesdk-integration

sethkfman committed:
fix: remove translations files being pushed up to crowdin (#6910)

sethkfman committed:
ci(action): unused unit testing segmentation scripts and yarn steps (#6422)

* increase the number of operations per run to 200 in-order to clear out historical backlog

* remove broken out unit test yarn steps

* remove unused unit test segmentation script

* Revert "increase the number of operations per run to 200 in-order to clear out historical backlog"

This reverts commit 9287b741edebdaba4ec519eb273b1183555e3a09.

sethkfman committed:
fix: remove unused code to overwrite filenames for Sentry error reporting (#6945)

* remove unused code to overwrite filenames for Sentry error reporting

* updated with comments and removed console.log

* remove unused function that was initally implemented so extension could have a single bundle

* remove console.log

elefantel committed:
Merge branch 'feature/android-nativesdk-integration' of github.com:MetaMask/metamask-mobile into feature/android-nativesdk-integration

abretonc7s committed:
fix: renamed sdk method

abretonc7s committed:
chore: linting

elefantel committed:
Merge branch 'feature/android-nativesdk-integration' of github.com:MetaMask/metamask-mobile into feature/android-nativesdk-integration

abretonc7s committed:
feat: avoid ios initialization

elefantel committed:
Merge branch 'feature/android-nativesdk-integration' of github.com:MetaMask/metamask-mobile into feature/android-nativesdk-integration

elefantel committed:
Update native sdk dependency

elefantel committed:
Merge branch 'main' into feature/android-nativesdk-integration

elefantel committed:
Remove console logs

elefantel committed:
Clean up debug logs

elefantel committed:
Migrate console logs to logger

omridan159 commented on app/components/Views/SDKSessionsManager/SDKSessionsManager.tsx:
```diff
@@ -83,9 +86,13 @@ const SDKSessionsManager = (props: Props) => {
-    const refreshSDKState = () => {
+    const refreshSDKState = async () => {
       const _connections = sdk.getConnections();
+      const _androidConnections = await sdk.loadAndroidConnections();
       const connectionsList = Object.values(_connections);
+      const androidConnectionsList = Object.values(_androidConnections);
+
+      setAndroidConnections(androidConnectionsList);
```
Consider adding error handling around this asynchronous call to manage potential failures.

```suggestion
      try {
        const _androidConnections = await sdk.loadAndroidConnections();
        const androidConnectionsList = Object.values(_androidConnections);
        setAndroidConnections(androidConnectionsList);
      } catch (error) {
        console.error("Failed to load Android connections:", error);
      }
```

elefantel committed:
Format logging

elefantel commented on app/components/Views/SDKSessionsManager/SDKSessionsManager.tsx:
```diff
@@ -83,9 +86,13 @@ const SDKSessionsManager = (props: Props) => {
       const connectionsList = Object.values(_connections);
+      const androidConnectionsList = Object.values(_androidConnections);
+
+      setAndroidConnections(androidConnectionsList);
```
Agreed! Applied a try-catch block here!

sethkfman committed:
ci(code-cov): adding code-cov to project (#6789)

* added code cov action

* update ignore patterns for test coverage

* add codecov.yml file

* update invalid codecov.yml file

* added range and percision codecov.yml file

* updated excluded files

* add ci pass required

* bump threshold

* added carriage return to codecov config file

* updated global coverage

* remove codecov

* setup config and baseline

* update threshold

* Update .github/workflows/ci.yml

Co-authored-by: Mark Stacey <markjstacey@gmail.com>

* remove token due to public repo

* fix: #893 fix immediate lock timer (#6653)

* Stage saga changes

* Create action creators for authentication state machine

* Remove unused code

* Start and end state machine

* Clean up LockScreen

* Clean up LockManager

* Remove unused code

* Notify auth state machine from Authentication service

* Organize auth state machine in sagas

* Change sagas to TS

* Remove unused changes

* Add type to navigation in NavigationService

* Clean up sagas

* Remove console log

* Add tests for sagas

* Remove password set

* Fix saga lint issues

* Fix missing deps

* Remove ts ignore from sagas

* Clarify navigation action

* Remove commented logger

* Create LOCK_SCREEN route const

* Provide comments to sagas

* Rename biometrics listener

* set timer immediately v2

* Add biometricsStateMachineId to auth code

* Remove unused user actions

* Clean up LockManager logic to work with Android and iOS

* Better handle authentication sagas

* Fix saga export error

* Fix saga unit tests

* Bump version for test

* Refine logout state when using biometrics

* Bump version

* Remove retry logic for lock screen

* Cancel biometrics state machine when logging out

* Clean up lock screen

* Reset isAuthenticating when biometrics fail in SecureKeychain.js

* Move LockScreen to top nav level

* Persist app state when LockManager shows.

* Fix saga tests

* Move LockManager initialization

* Use existing login and logout actions for biometric state machine

* Bump versions

* Use replace when logging out

* Update LockScreen snapshoit

* Comment bioStateMachineId

* Remove LockManager to LockManagerService

* Add comments to Authentication service

* Convert LockManagerService to Typescript

* Fix typo

* Update LockManagerService visibility modifiers

* Fix lint issues

---------

Co-authored-by: tommasini <46944231+tommasini@users.noreply.github.com>
Co-authored-by: tommasini <tommasini15@gmail.com>

* 7.4.0 (#6806)

* 7.4.0

* updated format changelog

* fix: define ReactNativePayments podspec path in podfile (#6821)

* from address changing on transaction object (#6841)

* fix: invalid transaction data used for approve transaction (#6833) (#6842)

* use valid txn data when approving txn

* rename variable

Co-authored-by: Sylva Elendu <iamsylvaelendu@gmail.com>

* add check approval request exists before rejecting (#6838)

* build 1155 RC2 (#6861)

* fix(test): Permission Systems fix taping connect modals (#6868)

* fix tapping connect modals

* fixed lint issues

* removed unused function, not referenced anywhere

* chore: merge 7.3.1 cherry-pick (#6923)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

* chore: version bump for 7.4.0 RC 3 (#6924)

* build 1159 RC4 (#6928)

---------

Co-authored-by: metamaskbot <metamaskbot@users.noreply.github.com>
Co-authored-by: sethkfman <Seth.Kaufman@consensys.net>
Co-authored-by: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Co-authored-by: tommasini <46944231+tommasini@users.noreply.github.com>
Co-authored-by: sethkfman <10342624+sethkfman@users.noreply.github.com>
Co-authored-by: Sylva Elendu <iamsylvaelendu@gmail.com>
Co-authored-by: Vinicius Stevam <45455812+vinistevam@users.noreply.github.com>
Co-authored-by: Chris Wilcox <chris.wilcox@consensys.net>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

* chore(deps): bump word-wrap from 1.2.3 to 1.2.4 (#6840)

Bumps [word-wrap](https://github.com/jonschlinkert/word-wrap) from 1.2.3 to 1.2.4.
- [Release notes](https://github.com/jonschlinkert/word-wrap/releases)
- [Commits](https://github.com/jonschlinkert/word-wrap/compare/1.2.3...1.2.4)

---
updated-dependencies:
- dependency-name: word-wrap
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: sethkfman <10342624+sethkfman@users.noreply.github.com>
Co-authored-by: legobeat <109787230+legobeat@users.noreply.github.com>

* chore: remove unused Jest preprocessor (#6913)

The file `jest.preprocessor.js` has been unused since #6170.

* docs: Update QA section in PR template (#6930)

* fix: Remove splash animation wait step from Cold Start Launch time test script (#6931)

* Change wait for splash animation to expect to displayed assertion

* Remove splash animation steps

* feat: Translation (#6937)

* remove onBackdropPress for signatures (#6936)

* feat: Migrate wdio 'SendEthMultisig' test to Detox (#6934)

* feat: github action to check if PR has requested labels before being merged (#6794)

* feat(action): check if pr includes requested labels

* fix(template): adjust PR temaple

* fix(action): add missing QA label

* fix(action): check if no label prevents merging

* fix(action): remove QA label check

* fix(action): add missing reopened condition

* fix(action): increase list of labels which prevent merges

* fix(action): add labeling guidelines

* fix(action): typo

* fix(action): typo

* fix: update README.md (#6853)

* Restore deleted SendEthMultisig.feature file (#6956)

* fix: custom position logic for badgeWrapper (#6864)

* Fixed custom position logic for badgeWrapper

* Fixed lint error

* Added README to Avatar (#6895)

* Added README to Avatar

* Updated readme based on comments

* Fixed reference link for AvatarAccount

* style: Updated minor checkbox style (#6942)

* Updated minor checkbox changes

* Updated MultiSelect snapshot

* feat: add support for template and header on result pages (#6832)

* Fix: add missed parenheses in the function call (#6958)

* translation ens ipfs alert (#6966)

* chore: small refactor (#6963)

* small refacto

* fix(): fixed linter errors

* fix(): fixed linter errors

---------

Co-authored-by: sethkfman <10342624+sethkfman@users.noreply.github.com>

* feat: blockaid what's new popup (#6901)

* Add blockaid what's new

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* New image for both dark/light mode

* image should be first
fix podfile

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* What's new fix navigation

* Revert Podfile

* use Routes

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* revert podfile.lock

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* Show what's new only if blockaid is enabled in settings.

---------

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* feat: Add security alerts settings to experimental tab (#6779)

* Add security alerts settings to experimental tab

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* add more providers label

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* change styles label

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* Fix tests

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* Sort imports

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* use environment var to determine if to show blockaid or not

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* Removed Podfile changes

---------

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* updated jest config to have an allow list instead of eclude list for test coverage

* bumped thresholds

* added exclusions for mocks, util/test, util/testUtils

---------

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>
Co-authored-by: Mark Stacey <markjstacey@gmail.com>
Co-authored-by: Cal Leung <cleun007@gmail.com>
Co-authored-by: tommasini <46944231+tommasini@users.noreply.github.com>
Co-authored-by: tommasini <tommasini15@gmail.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: metamaskbot <metamaskbot@users.noreply.github.com>
Co-authored-by: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Co-authored-by: Sylva Elendu <iamsylvaelendu@gmail.com>
Co-authored-by: Vinicius Stevam <45455812+vinistevam@users.noreply.github.com>
Co-authored-by: Chris Wilcox <chris.wilcox@consensys.net>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: legobeat <109787230+legobeat@users.noreply.github.com>
Co-authored-by: Gustavo Antunes <17601467+gantunesr@users.noreply.github.com>
Co-authored-by: SamuelSalas <samuel.salas.reyes@gmail.com>
Co-authored-by: sleepytanya <104780023+sleepytanya@users.noreply.github.com>
Co-authored-by: Gauthier Petetin <gauthierpetetin@hotmail.com>
Co-authored-by: Brian August Nguyen <brianacnguyen@gmail.com>
Co-authored-by: Menox <marcomenozzi80@gmail.com>
Co-authored-by: Olusegun Akintayo <trinisoftinc@gmail.com>

omridan159 commented on app/core/SDKConnect/AndroidSDK/AndroidService.ts:
```diff
@@ -0,0 +1,387 @@
+    // Bind to native module
+    await SDKConnect.getInstance().bindAndroidSDK();
+
+    if (this.connectedClients) {
```
I think that checks like `if (this.connectedClients)` may not be necessary because `this.connectedClients` is always defined as an object during the class instantiation and by that always truthy unless in some case you set it to undefined

Maybe this would be a better check `Object.keys(this.connectedClients ?? {}).length`

elefantel replied:
Perhaps `Object.keys(this.connectedClients).length` should suffice. Do we need the `?? {}`?

omridan159 replied:
Just in case it would be undefined in some case, and then we will get a JS error.
But in general it's ok not to put the `?? {}`, but I think it's better to put it there just for safety.

omridan159 commented on app/core/SDKConnect/utils/wait.util.ts:
```diff
@@ -1,11 +1,31 @@
     setTimeout(resolve, ms);
   });
 
+export const waitForReadyClient = async (
+  id: string,
+  connectedClients: {
+    [clientId: string]: AndroidClient;
+  },
+) => {
+  let i = 0;
+  while (!connectedClients[id]) {
+    i += 1;
+    if (i++ > MAX_QUEUE_LOOP) {
+      console.warn(`RPC queue not empty after ${MAX_QUEUE_LOOP} seconds`);
+      break;
+    }
+    await wait(1000);
+  }
+};
```
```suggestion
export const waitForReadyClient = async (
  id: string,
  connectedClients: {
    [clientId: string]: AndroidClient;
  },
) => {
  let attempts = 0;
  while (!connectedClients[id]) {
    attempts++;
    if (attempts > MAX_QUEUE_LOOP) {
      console.warn(`RPC queue not empty after ${MAX_QUEUE_LOOP} seconds`);
       break;
    }
    await wait(1000);
  }
};
```

More readable

jpcloureiro committed:
fix: vault recovery & invalid password error (#6957)

* ResetPassword: remove unused redux action passwordUnset

* Authentication: set `state.user.passwordSet` boolean after successful auth

* Vault Recovery: reset engine initialization

set EngineService engineInitialized instance to false to ensure we
dispatch INIT_BG_STATE_KEY. This will save KeyringController state object
to be saved into redux store state

* EngineService: add unit tests

---------

Co-authored-by: Owen Craston <owengc12@gmail.com>
Co-authored-by: Cal Leung <cleun007@gmail.com>

elefantel committed:
Remove AndroidServicePorts

elefantel committed:
Clean up commemts

elefantel committed:
Restructure event listeners

abretonc7s committed:
fix: linting issue with async promise handler

elefantel committed:
Merge branch 'main' into feature/android-nativesdk-integration

andreahaku reviewed (dismissed):
LGTM

elefantel committed:
Use a release version of the native sdk

andreahaku reviewed (approved):
LGTM
